API/Collection/ObjectStateCollection.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Kaliop\eZMigrationBundle\API\Collection;
+
+/**
+ * @todo add phpdoc to suggest typehinting
+ */
+class ObjectStateCollection extends AbstractCollection
+{
+    protected $allowedClass = 'eZ\Publish\API\Repository\Values\ObjectState\ObjectState';
```

Comment:
For better usability, I'd use `::class` to get the right class, as it allows click through:
```php
protected $allowedClass = \eZ\Publish\API\Repository\Values\ObjectState\ObjectState::class;
```
(unless some specific eZ things prevent from it)